Makefile

```diff
@@ -48,8 +58,8 @@ else
     $(error "build option 'DEBUG' must be 'release', 'debug' or 'maintainer'")
 endif
 
-LIB_DYNAMIC_PATH=$(CARGO_TARGET_DIR)/$(VARIANT_NAME)/$(LIB_NAME).$(DYNLIB_EXT)
-LIB_STATIC_PATH=$(CARGO_TARGET_DIR)/$(VARIANT_NAME)/$(LIB_NAME).a
+LIB_DYNAMIC_PATH=$(CARGO_TARGET_DIR)/$(ARCH_NAME)/$(VARIANT_NAME)/$(LIB_NAME).$(DYNLIB_EXT)
+LIB_STATIC_PATH=$(CARGO_TARGET_DIR)/$(ARCH_NAME)/$(VARIANT_NAME)/$(LIB_NAME).a
```

Comment:
ARCH_NAME isn't set if this is not MacOS. Do we want to set it to a default value?

Reply:
That is a point, but I'd prefer to leave it as-is for now, it builds and tests on ubuntu.